mlperf_logging/result_summarizer/result_summarizer.py

```diff
@@ -159,131 +197,314 @@ def _is_organization_folder(folder):
+            for metric, dtype in _get_weak_scaling_metric_schema().items():
+                schema['{}:{}'.format(benchmark, metric)] = dtype
+    else:
+        schema.update(
+            {b: float
+             for b in get_allowed_benchmarks(usage, ruleset)})
+    schema.update({'details_url': str, 'code_url': str})
+    return schema
+
+
+class FieldError(ValueError):
+    pass
+
+
+def _assert_in_desc_and_return(desc, desc_keys, query=None):
+    if not isinstance(desc_keys, (list, tuple, set)):
+        desc_keys = (desc_keys, )
+    if query is None:
+        assert len(desc_keys) == 1
+    for desc_key in desc_keys:
+        if desc_key not in desc:
+            raise FieldError('ERROR: "{}" field missing'.format(desc_key))
+    return desc[desc_keys[0]] if query is None else query(desc)
+
+
+def _compute_strong_scaling_scores(desc, system_folder, usage, ruleset):
+    # Collect scores for benchmarks.
+    benchmark_scores = {}
+    benchmark_folder_parent = os.path.join(
+        system_folder, 'strong') if usage == 'hpc' else system_folder
+    if not os.path.isdir(benchmark_folder_parent):
+        return benchmark_scores
+    for benchmark_folder in _get_sub_folders(benchmark_folder_parent):
+        folder_parts = benchmark_folder.split('/')
+        benchmark = _benchmark_alias(folder_parts[-1])
+        system = folder_parts[-3] if usage == 'hpc' else folder_parts[-2]
+        # Read scores from result files.
+        pattern = '{folder}/result_*.txt'.format(folder=benchmark_folder)
+        result_files = glob.glob(pattern, recursive=True)
+        scores = []
+        dropped_scores = 0
+        for result_file in result_files:
+            try:
+                loglines = _read_result_file(result_file, usage, ruleset)
+                scores.append(_query_mlperf_strong_scaling_score(loglines))
+            except ValueError as e:
+                print('{} in {}'.format(e, result_file))
+                dropped_scores += 1
+                continue
+        max_dropped_scores = 4 if benchmark == 'unet3d' else 1
+        if dropped_scores > max_dropped_scores:
+            print('CRITICAL ERROR: Too many non-converging runs '
+                  'for {} {}/{}'.format(desc['submitter'], system, benchmark))
+            print('** CRITICAL ERROR ** Results in the table for {} {}/{} are '
+                  'NOT correct'.format(desc['submitter'], system, benchmark))
+        elif dropped_scores >= 1:
+            print('NOTICE: Dropping non-converged run(s) for {} {}/{} using '
+                  'olympic scoring.'.format(
+                      desc['submitter'],
+                      system,
+                      benchmark,
+                  ))
+
+        if dropped_scores <= max_dropped_scores:
+            benchmark_scores[benchmark] = _compute_olympic_average(
+                scores, dropped_scores, max_dropped_scores)
+
+    _fill_empty_benchmark_scores(benchmark_scores, usage, ruleset)
```

Comment:
What does this do exactly?

Reply:
For each system, a submitter might not submit results for all benchmarks. When that happens, `_fill_empty_benchmark_scores` fill in `None`s for the benchmark for which no result was submitted, and it would show up as filler values (i.e., NaN) in the result summary.